### PR TITLE
Improve CLI

### PR DESCRIPTION
**Pull request by Holzhaus**
Based on top of PR #81.

This switches to a single binary and uses clap for a nicer command line interface.
It also adds support for a new list-playlists command that i implemented while debugging https://github.com/mixxxdj/mixxx/issues/10923#issuecomment-1270307397.

```
$ cargo run -- --help
Library for parsing Pioneer Rekordbox device exports

Usage: rekordcrate <COMMAND>

Commands:
  list-playlists  List the playlist tree from a Pioneer Database (`.PDB`) file
  dump-anlz       Parse and dump a Rekordbox Analysis (`ANLZXXXX.DAT`) file
  dump-pdb        Parse and dump a Pioneer Database (`.PDB`) file
  dump-setting    Parse and dump a Pioneer Settings (`*SETTING.DAT`) file
  help            Print this message or the help of the given subcommand(s)

Options:
  -h, --help     Print help information
  
$ cargo run -- list-playlists export.pdb  | head -n 10
🗀 15.07.22
    🗎 riot code
🗀 jaki_21_05_22
    🗎 jaki_melow
    🗎 jaki2
    🗎 jaki
🗀 18.05.22
    🗎 plaid - dial p
    🗎 otherstuff
    🗎 wettrax
  ```

### CODE REVIEW

**Comment by Holzhaus** [comment]
We may consider splitting out the binary from the library later on, as cargo doesn't support binary-specific dependencies at the moment and its a bit odd to have clap as library dependency.

**Comment by Swiftb0y** [comment]
Mhmm yeah. I'm not nearly familiar enough with cargo and its features to the point where I can comment on that though. But it doesn't seem that odd of a feature, there must be an RFC or at least a feature request on this somewhere.

**Comment by Holzhaus** [comment]
> Mhmm yeah. I'm not nearly familiar enough with cargo and its features to the point where I can comment on that though. But it doesn't seem that odd of a feature, there must be an RFC or at least a feature request on this somewhere.

Yup, there is https://github.com/rust-lang/cargo/issues/1982.

Nothing to worry about now. we can reoganize the repo into a cargo workspace and  split the code into a `rekordcrate` and `rekordcrate-bin` crate later if we think it's necessary.

File: src/main.rs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        #[arg(value_name = "SETTING_FILE")]
+        path: PathBuf,
+    },
+}
+
+fn list_playlists(path: &PathBuf) {
+    use rekordcrate::pdb::PlaylistTreeNode;
+    use std::collections::HashMap;
+
+    fn print_children_of(tree: &HashMap<u32, Vec<PlaylistTreeNode>>, id: &u32, level: usize) {
```
Comment (Swiftb0y) [inline src/main.rs:56]:
```suggestion
    fn print_children_of(tree: &HashMap<u32, Vec<PlaylistTreeNode>>, id: u32, level: usize) {
```
why pass the `Copy` type as ref?

**Review by Swiftb0y — CHANGES_REQUESTED** [review]
lgtm otherwise

File: src/main.rs (hunk 1 above)
Comment (Holzhaus) [reply src/main.rs]: https://github.com/Holzhaus/rekordcrate/pull/82/commits/1fd43c118553d1ff18abe52b3bb8272069d69aea

**Pull request merged**